File: Server/src/main/java/com/api/server/foodb/FoodcomexVendorCompound.java
package com.api.server.foodb;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "foodcomex_vendor_compounds")
public class FoodcomexVendorCompound {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @Size(max = 255)
    @NotNull
    @Column(name = "foodcomex_vendor_id", nullable = false)
    private String foodcomexVendorId;

    @Size(max = 255)
    @Column(name = "catalogue_number")
    private String catalogueNumber;

    @Lob
    @Column(name = "name")
    private String name;

    @Size(max = 255)
    @Column(name = "cas_number")
    private String casNumber;

    @Size(max = 255)
    @Column(name = "inchikey")
    private String inchikey;

    @Size(max = 255)
    @Column(name = "url")
    private String url;

    @NotNull
    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @NotNull
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

}

File: DotNetServer/DotNetServer/Core/Entities/Foodb/FoodcomexVendorCompound.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace DotNetServer.Core.Entities.Foodb;

[Table("foodcomex_vendor_compounds")]
[Index("CasNumber", Name = "index_foodcomex_vendor_compounds_on_cas_number")]
[Index("Inchikey", Name = "index_foodcomex_vendor_compounds_on_inchikey")]
public partial class FoodcomexVendorCompound
{
    [Key] [Column("id")] public int Id { get; set; }

    [Column("foodcomex_vendor_id")]
    [StringLength(255)]
    public string FoodcomexVendorId { get; set; } = null!;

    [Column("catalogue_number")]
    [StringLength(255)]
    public string? CatalogueNumber { get; set; }

    [Column("name", TypeName = "text")] public string? Name { get; set; }

    [Column("cas_number")] public string? CasNumber { get; set; }

    [Column("inchikey")] public string? Inchikey { get; set; }

    [Column("url")] [StringLength(255)] public string? Url { get; set; }

    [Column("created_at", TypeName = "datetime")]
    public DateTime CreatedAt { get; set; }

    [Column("updated_at", TypeName = "datetime")]
    public DateTime UpdatedAt { get; set; }
}